libs/web3-api-provider/src/webb-provider/vanchor-withdraw.ts

```diff
@@ -420,16 +386,39 @@ export class Web3VAnchorWithdraw extends VAnchorWithdraw<WebbWeb3Provider> {
           this.inner.noteManager?.removeNote(parsedNote);
         }
       } else {
-        const tx = await destVAnchor.inner.transact(
-          {
-            ...publicInputs,
-            outputCommitments: [
-              publicInputs.outputCommitments[0],
-              publicInputs.outputCommitments[1],
-            ],
-          },
-          extData
-        );
+        let tx: ContractTransaction;
+
+        if (unwrapTokenAddress) {
+          tx = await destVAnchor.inner.transactWrap(
+            {
+              ...publicInputs,
+              outputCommitments: [
+                publicInputs.outputCommitments[0],
+                publicInputs.outputCommitments[1],
+              ],
+            },
+            extData,
+            unwrapTokenAddress,
+            {
+              gasLimit: 10000000 * 9,
```

Comment:
@dutterbutter  can we have a task for improving gas limit calculation?

Reply:
Gas limit is fixed on these chains so really no need for calculation. Just need to keep a map and update it from time to time (these things update infrequently)

Reply:
Gas price is what changes and could be calculated